runelite-client/src/main/java/net/runelite/client/plugins/boosts/BoostsOverlay.java

```diff
@@ -72,59 +75,97 @@ public Dimension render(Graphics2D graphics, Point parent)
 			return null;
 		}
 
-		Skill[] show;
-		if (config.enableSkill())
-		{
-			show = ObjectArrays.concat(COMBAT, SKILLING, Skill.class);
-		}
-		else
-		{
-			show = COMBAT;
-		}
-
 		panelComponent = new PanelComponent();
 
-		for (Skill skill : show)
+		for (Skill skill : plugin.getShownSkills())
 		{
 			int boosted = client.getBoostedSkillLevel(skill),
 				base = client.getRealSkillLevel(skill);
 
 			if (boosted == base)
 			{
+				if (indicators[skill.ordinal()] != null)
+				{
+					infoBoxManager.removeInfoBox(indicators[skill.ordinal()]);
+					indicators[skill.ordinal()] = null;
+				}
+
 				continue;
 			}
 
-
-			String str;
-			Color strColor = Color.WHITE;
-			if (!config.useRelativeBoost())
+			if (config.displayIndicators())
 			{
-				str = boosted + "/" + base;
+				if (indicators[skill.ordinal()] == null)
+				{
+					BoostIndicator indicator = new BoostIndicator(skill, getSkillImage(skill), client, config);
+					indicators[skill.ordinal()] = indicator;
+					infoBoxManager.addInfoBox(indicator);
+				}
 			}
 			else
 			{
-				int boost = boosted - base;
-				str = String.valueOf(boost);
-				if (boost > 0)
+				String str;
+				Color strColor = Color.WHITE;
+				if (!config.useRelativeBoost())
 				{
-					str = "+"  + str;
-					strColor = Color.GREEN.darker();
+					str = boosted + "/" + base;
 				}
 				else
 				{
-					strColor = Color.RED.darker();
+					int boost = boosted - base;
+					str = String.valueOf(boost);
+					strColor = getTextColor(boost);
+					if (boost > 0)
+					{
+						str = "+"  + str;
+					}
 				}
+
+				panelComponent.getLines().add(new PanelComponent.Line(
+						skill.getName(),
+						Color.WHITE,
+						str,
+						strColor
+				));
 			}
+		}
 
-			panelComponent.getLines().add(new PanelComponent.Line(
-				skill.getName(),
-				Color.WHITE,
-				str,
-				strColor
-			));
+		return config.displayIndicators() ? null : panelComponent.render(graphics, parent);
+	}
+
+	private Color getTextColor(int boost)
+	{
+		if (boost > 0)
+		{
+			return Color.GREEN;
 		}
 
-		return panelComponent.render(graphics, parent);
+		return new Color(238, 51, 51);
+
 	}
 
+	private BufferedImage getSkillImage(Skill skill)
```

Comment:
Maybe this skill icon logic should move somewhere centrally since its starting to appear many places?